include/openthread/link.h

```diff
@@ -216,6 +217,34 @@ typedef struct otMacCounters
      */
     uint32_t mTxRetry;
 
+    /**
+     * Histogram of number of retries for a single direct packet until success
+     * [0 retry: packet count, 1 retries: packet count, 2 retries : packet count, ...,
+     * n retries : packet count (n == max retry limit)]
+     * The size of the array is OPENTHREAD_CONFIG_MAC_DEFAULT_MAX_FRAME_RETRIES_DIRECT.
+     */
+    uint32_t mTxDirectRetrySuccess[OPENTHREAD_CONFIG_MAC_DEFAULT_MAX_FRAME_RETRIES_DIRECT];
```

Comment:
We have recent APIs which allow user to change the retry numbers from at run-time `otLinkSetMaxFrameRetriesDirect()` and `otLinkSetMaxFrameRetriesIndirect()` from https://github.com/openthread/openthread/pull/4256 

`OPENTHREAD_CONFIG_MAC_DEFAULT_MAX_FRAME_RETRIES_DIRECT` is the default value but with the new API the user can change the retry count from default to a different and possibly larger value.

We need a solution on how to handle this. 
- Do we define an array of counters with some max length and ignore if user sets retry count larger than our array size?
- If user changes the retry count, do we reset the histogram counters?